ablySpec/Auth.swift

```diff
@@ -529,5 +529,256 @@ class Auth : QuickSpec {
+                expect(tokenDetails?.expires?.timeIntervalSince1970).to(beGreaterThan(tokenDetails?.issued?.timeIntervalSince1970))
+            }
+
+            // RSA10f
+            it("should return TokenDetails with valid token metadata") {
+                let options = AblyTests.commonAppSetup()
+                options.clientId = "client_string"
+                let rest = ARTRest(options: options)
+
+                waitUntil(timeout: testTimeout) { done in
+                    rest.auth.authorise(nil, options: nil) { tokenDetails, error in
+                        expect(error).to(beNil())
+                        expectTokenDetails(tokenDetails)
+                        expect(tokenDetails?.clientId).to(equal(options.clientId))
+                        done()
+                    }
+                }
+            }
+
+            // RSA10h
+            it("should use the configured Auth#clientId, if not null, by default") {
+                let options = AblyTests.commonAppSetup()
+
+                // ClientId null
+                waitUntil(timeout: testTimeout) { done in
+                    ARTRest(options: options).auth.authorise(nil, options: nil) { tokenDetails, error in
+                        expect(error).to(beNil())
+                        expect(tokenDetails).toNot(beNil())
+                        expect(tokenDetails?.clientId).to(beNil())
+                        done()
+                    }
+                }
+
+                options.clientId = "client_string"
+
+                // ClientId not null
+                waitUntil(timeout: testTimeout) { done in
+                    ARTRest(options: options).auth.authorise(nil, options: nil) { tokenDetails, error in
+                        expect(error).to(beNil())
+                        expect(tokenDetails).toNot(beNil())
+                        expect(tokenDetails?.clientId).to(equal(options.clientId))
+                        done()
+                    }
+                }
+            }
+
+            // RSA10i
+            context("should adhere to all requirements relating to") {
+
+                it("TokenParams and authCallback") {
+                    let options = AblyTests.commonAppSetup()
+                    options.clientId = "client_string"
+                    let rest = ARTRest(options: options)
+
+                    let expectedClientId = "client_from_callback"
+
+                    let tokenParams = ARTAuthTokenParams()
+                    tokenParams.clientId = "client_from_params"
+
+                    let authOptions = ARTAuthOptions()
+                    authOptions.authCallback = { tokenParams, callback in
+                        rest.auth.createTokenRequest(ARTAuthTokenParams(clientId: expectedClientId), options: options, callback: callback)
+                    }
+
+                    waitUntil(timeout: testTimeout) { done in
+                        rest.auth.authorise(tokenParams, options: authOptions) { tokenDetails, error in
```

Comment:
> should be initialised with only a token request callback and no API key

@mattheworiordan it gives an error: `Error Domain=io.ably.cocoa Code=80001 "API Key is missing" UserInfo={NSLocalizedDescription=API Key is missing}`

Am I missing something?
Code:
``` swift
let rest = ARTRest(options: AblyTests.commonAppSetup())

var tokenRequest: ARTAuthTokenRequest? = nil
rest.auth.createTokenRequest(nil, options: nil, callback: { currentTokenRequest, error in
    tokenRequest = currentTokenRequest
})
expect(tokenRequest).toEventuallyNot(beNil(), timeout: testTimeout)

let options = ARTClientOptions()
options.authCallback = { tokenParams, callback in
    return tokenRequest
}

ARTRest(options: options).auth.authorise(nil, options: nil) { tokenDetails, error in
    expect(error).to(beNil()) //<-- error
    expectTokenDetails(tokenDetails)
}
```